functions/hierarchyCheck.js

```diff
@@ -0,0 +1,44 @@
+/**
+Check if user is higher in guild's role hierarchy
+...Basically an easy way to stop mods from performing actions on admins, etc
+
+    Usage:
+        - client scope
+        - executor is the user object of the user executing the moderation command
+        - target is the user object of the user the moderation command is affecting
+        - guild is the guild object of the guild the command is being executed in
+    Returns:
+        - boolean true if executor is higher
+        - boolean false if executor is lower or equal
+        - boolean false if either user isn't in the guild
+
+        **/
+
+const func = async (client, executor, target, guild = null) => {
+  try {
```

Comment:
I put the try/catch in because (despite this being stupid, and something most people don't do) I've seen lots of people already try and use moderation commands on users not in their guild by manually tagging them (ie: manually typing `<@userID>` in the client). In the event of it happening, I prefer to have the command notify them of the failure, rather than just crash and leave a (seemingly) unresolved command in chat.
![screenshot from 2017-06-14 00-08-45](https://user-images.githubusercontent.com/16696023/27119719-cff550ca-5095-11e7-844c-2d84e6db1fbe.png)